include/unifex/task.hpp

```diff
@@ -155,12 +174,85 @@ struct _promise {
       this->expected_.state_ = _state::exception;
     }
 
-    template(typename Value)
-        (requires callable<decltype(unifex::await_transform), type&, Value>)
-    auto await_transform(Value&& value)
-        noexcept(is_nothrow_callable_v<decltype(unifex::await_transform), type&, Value>)
-        -> callable_result_t<decltype(unifex::await_transform), type&, Value> {
-      return unifex::await_transform(*this, (Value&&)value);
+    template <typename Value>
+    decltype(auto) await_transform(Value&& value) {
+      if constexpr (derived_from<remove_cvref_t<Value>, _task_base>) {
+        // We are co_await-ing a unifex::task or something that completes inline. We don't
+        // need an additional transition.
```

Comment:
This comment is out of date.
```suggestion
        // We are co_await-ing a unifex::task. We don't need an additional transition.
```